pilot/pkg/serviceregistry/kube/controller/ambient/helpers.go

```diff
@@ -65,21 +67,59 @@ func serviceToAddressInfo(s *workloadapi.Service) model.AddressInfo {
 	}
 }
 
-func byteIPToString(b []byte) string {
-	ip, _ := netip.AddrFromSlice(b)
+func mustByteIPToString(b []byte) string {
+	ip, _ := netip.AddrFromSlice(b) // Address only comes from objects we create, so it must be valid
 	return ip.String()
 }
 
 func byteIPToAddr(b []byte) netip.Addr {
-	ip, _ := netip.AddrFromSlice(b)
+	ip, _ := netip.AddrFromSlice(b) // Address only comes from objects we create, so it must be valid
 	return ip
 }
 
-func (a *index) toNetworkAddress(vip string) *workloadapi.NetworkAddress {
+func (a *index) toNetworkAddress(vip string) (*workloadapi.NetworkAddress, error) {
+	ip, err := netip.ParseAddr(vip)
+	if err != nil {
+		return nil, fmt.Errorf("parse %v: %v", vip, err)
+	}
+	return &workloadapi.NetworkAddress{
+		Network: a.Network(vip, make(labels.Instance, 0)).String(),
+		Address: ip.AsSlice(),
+	}, nil
+}
+
+func (a *index) toNetworkAddressFromIP(ip netip.Addr) *workloadapi.NetworkAddress {
+	return &workloadapi.NetworkAddress{
+		Network: a.Network(ip.String(), make(labels.Instance, 0)).String(),
+		Address: ip.AsSlice(),
+	}
+}
+
+func (a *index) toNetworkAddressFromCidr(vip string) (*workloadapi.NetworkAddress, error) {
```

Comment:
this is used to parse address from SE, which is a valid CIDR. Do we really need to check error again

Reply:
it's not any cidr that is allowed, see my other comment